ql/src/test/queries/clientpositive/materialized_view_rewrite_by_text_10.q

```diff
@@ -0,0 +1,11 @@
+-- Materialzed view definition has non-deterministic function
+set hive.support.concurrency=true;
+set hive.txn.manager=org.apache.hadoop.hive.ql.lockmgr.DbTxnManager;
+
+CREATE TABLE EMPS (ENAME STRING, BIRTH_EPOCH_SECS INT) STORED AS ORC TBLPROPERTIES ('transactional'='true');
+
+CREATE MATERIALIZED VIEW v_emp AS SELECT * FROM EMPS WHERE BIRTH_EPOCH_SECS <= UNIX_TIMESTAMP();
```

Comment:
Shouldn't the CREATE fail completely? I mentioned this in the code as well but I don't understand why CREATE and ALTER behave differently.

Reply:
Actually thinking a bit more on this, it may be fine to allow the creation of the materialized view as long as it does not create issues with query rewriting and (incremental) refresh.

Reply:
Yes. Creation should not fail but a message is logged and MV is disabled for rewrite if none of the algorithms are available.

In case of Alter the goal is to enable rewrite but if none of the algorithms are available I think it is ok to fail.